pkg/controller/elasticsearch/services/services.go

```diff
@@ -67,19 +68,60 @@ func ExternalServiceName(esName string) string {
 	return esv1.HTTPService(esName)
 }
 
+// InternalServiceName returns the name for the internal service
+// associated to this cluster, managed by the operator exclusively.
+func InternalServiceName(esName string) string {
+	return esv1.InternalHTTPService(esName)
+}
+
 // ExternalTransportServiceHost returns the hostname and the port used to reach Elasticsearch's transport endpoint.
 func ExternalTransportServiceHost(es types.NamespacedName) string {
 	return stringsutil.Concat(TransportServiceName(es.Name), ".", es.Namespace, globalServiceSuffix, ":", strconv.Itoa(network.TransportPort))
 }
 
-// ExternalServiceURL returns the URL used to reach Elasticsearch's external endpoint
+// ExternalServiceURL returns the URL used to reach Elasticsearch's external endpoint.
 func ExternalServiceURL(es esv1.Elasticsearch) string {
 	return stringsutil.Concat(es.Spec.HTTP.Protocol(), "://", ExternalServiceName(es.Name), ".", es.Namespace, globalServiceSuffix, ":", strconv.Itoa(network.HTTPPort))
 }
 
-// NewExternalService returns the external service associated to the given cluster
+// InternalServiceURL returns the URL used to reach Elasticsearch's internally managed service
+func InternalServiceURL(es esv1.Elasticsearch) string {
+	return stringsutil.Concat(es.Spec.HTTP.Protocol(), "://", InternalServiceName(es.Name), ".", es.Namespace, globalServiceSuffix, ":", strconv.Itoa(network.HTTPPort))
+}
+
+// NewExternalService returns the external service associated to the given cluster.
 // It is used by users to perform requests against one of the cluster nodes.
 func NewExternalService(es esv1.Elasticsearch) *corev1.Service {
+	return newServiceWithName(es, ExternalServiceName(es.Name))
+}
+
+// NewInternalService returns the internal service associated to the given cluster.
+// It is used by the operator to perform requests against the Elasticsearch cluster nodes,
+// and does not inherit the spec defined within the Elasticsearch custom resource,
+// to remove the possibility of the user misconfiguring access to the ES cluster.
+func NewInternalService(es esv1.Elasticsearch) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      InternalServiceName(es.Name),
+			Namespace: es.Namespace,
+			Labels:    label.NewLabels(k8s.ExtractNamespacedName(&es)),
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{
+					Name:     es.Spec.HTTP.Protocol(),
+					Protocol: corev1.ProtocolTCP,
+					Port:     network.HTTPPort,
+				},
+			},
+			Selector:                 label.NewLabels(k8s.ExtractNamespacedName(&es)),
+			PublishNotReadyAddresses: false,
+		},
+	}
+}
+
+func newServiceWithName(es esv1.Elasticsearch, serviceName string) *corev1.Service {
```

Comment:
nit: this function seems to be used only in `NewExternalService` I would keep it inlined.

Reply:
makes sense. 👍